tests/Cache/ArrayResultTest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Cache;
+
+use Doctrine\DBAL\Cache\ArrayResult;
+use Doctrine\DBAL\Exception\InvalidColumnIndex;
+use PHPUnit\Framework\Attributes\TestWith;
+use PHPUnit\Framework\TestCase;
+
+class ArrayResultTest extends TestCase
```

Comment:
It's a bit unfortunate that git does not recognize that `ArrayStatementTest` has been renamed to `ArrayResultTest`. I'll try to backport that change before merging your PR.

Reply:
Okay, you've practically rewritten the whole test. Let's merge as is.